### DISCUSSION TITLE
A tool to recover forgotten passwords! (IN PYTHON!)

### DISCUSSION THREAD

**Discussion by NHbinaaa111** [community/community#46371] [Programming Help]
### Select Topic Area

Show & Tell

### Body

 
Hello World! 

I am quite new to the community as I joined github recently. I have been developing quite a project for the past few months, which is called HackItBack. It is a tool to recover partially forgotten passwords. It takes some assumption values and some known values and merges them together in a specific order and makes a list of random passwords (which have a chance to containing the original password) 

This list can be used to carry out dictionary attacks on your account which may unlock it. 

I have no proof until now if it works, but overall I think the idea is a bit stupid and I do not know if it is even possible. 

You can run the tool in the browser too:
https://replit.com/@NHbinaaa111/HackItBack?embed=true

Kindly give me feedback on what should I add or if it is trash. I am willing to accept brutally honest feedback too so feel free.

Edit:I think I messed up, these discussions are for asking questions only?

**Comment by jge162** [comment]
Hello @NHbinaaa111 ues this forum is a discussion form. You can ask questions and talk about relevant things. But, what you are working on good for you.best of luck.

**Comment by mammaddrik** [comment]
Hello, I think it's an interesting idea, even though it couldn't create or find my password, but it was found to be different by one or two numbers, And I suggest that you save the passwords in a file.

If you have such ideas and would like to have a better program, I definitely recommend this script:
GitHub [Here](https://github.com/Mebus/cupp).

By trying and spending time on programming, you can improve your program, so move forward to success.

**Reply by NHbinaaa111** [reply]
Thank you for the compliment!
I'm thinking of making it automated so that it makes a list of the passwords, makes it a file, stores it and then tries every single password on a specific login page.

**Comment by lbirkert** [comment]
> ### Select Topic Area
> 
> Show & Tell
> ### Body
> 
> Hello World!
> 
> I am quite new to the community as I joined github recently. I have been developing quite a project for the past few months, which is called HackItBack. It is a tool to recover partially forgotten passwords. It takes some assumption values and some known values and merges them together in a specific order and makes a list of random passwords (which have a chance to containing the original password)
> 
> This list can be used to carry out dictionary attacks on your account which may unlock it.
> 
> I have no proof until now if it works, but overall I think the idea is a bit stupid and I do not know if it is even possible.
> 
> You can run the tool in the browser too: https://replit.com/@NHbinaaa111/HackItBack?embed=true
> 
> Kindly give me feedback on what should I add or if it is trash. I am willing to accept brutally honest feedback too so feel free.
> 
> Edit:I think I messed up, these discussions are for asking questions only?

Hey,

this is an interesting idea but the major flaw is the fact that the quantity of factors you'd need to reconstruct a list that is under a certain length (perhaps under 500) is too high, which makes the tool unpractical in most scenarios.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @NHbinaaa111 ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by NHbinaaa111** [reply]
Thank you!

**Comment by mpboom** [comment]
That looks like a great start. Some general feedback would be:

- It's common to put imports at the top of the file
- You don't seem to be using a lot of data structures like lists and dictionaries, while they are very useful for these kinds of applications. Instead of having to create a variable for every character and limiting yourself to an amount of characters this way, you could try putting them in a list, set or dictionary.

**Reply by NHbinaaa111** [reply]
I appreciate the feedback, I will start implementing data structures in the later versions.
